R/infrastructure.R

```diff
@@ -301,30 +322,46 @@ use_package_doc <- function(pkg = ".") {
 #'   "Depends", "Suggests", "Enhances", or "LinkingTo" (or unique abbreviation)
 #' @param pkg package description, can be path or package name. See
 #'   \code{\link{as.package}} for more information.
+#' @param version Value indicating whether \code{package}'s version should be included
+#'   in \code{pkg}'s DESCRIPTION. Either \code{NULL}, \code{TRUE}, or a valid version
+#'   string. If \code{version} is \code{TRUE}, \code{package}'s current version
+#'   will be used (see examples). If \code{version} is a string that is not a
+#'   valid version, an error will be thrown.
+#' @param compare The comparator used for \code{package}'s version. All valid
+#'   CRAN comparators (i.e. \code{==}, \code{>=}, \code{<=}, \code{>}, and
+#'   \code{<}) are accepted, but not considered unless \code{version} is
+#'   non-\code{NULL}.
 #' @family infrastructure
 #' @export
 #' @examples
 #' \dontrun{
 #' use_package("ggplot2")
 #' use_package("dplyr", "suggests")
-#'
+#' use_package("devtools", version = TRUE) # "devtools (>= 1.10.0.9000)"
+#' use_package("devtools", version = TRUE, compare = "==") # "devtools (== 1.10.0.9000)"
+#' use_package("devtools", version = "1.10") # "devtools (>= 1.10)"
+#' use_package("devtools", version = "1.10", compare = ">") # "devtools (> 1.10)"
 #' }
-use_package <- function(package, type = "Imports", pkg = ".") {
+use_package <- function(package, type = "Imports", pkg = ".", version = NULL,
+                        compare =  c(">=", ">", "==", "<=", "<")) {
+
   stopifnot(is.character(package), length(package) == 1)
   stopifnot(is.character(type), length(type) == 1)
 
   if (!is_installed(package)) {
     stop(package, " must be installed before you can take a dependency on it",
-      call. = FALSE)
+         call. = FALSE)
   }
 
+  package_txt <- build_package_txt(package, version, compare)
```

Comment:
this is defined as close as I can make it (one line above `use_package()`'s roxumentation), while still making it accessible to the [unit tests](https://github.com/hadley/devtools/pull/1054/files#diff-67b82d9b2f120c72b1173ce45fcdb8e7R120)